R/tm_g_scatterplot.R

```diff
@@ -389,9 +388,60 @@ srv_g_scatterplot <- function(id,
   checkmate::assert_class(data, "tdata")
   moduleServer(id, function(input, output, session) {
     data_extract <- list(
-      x = x, y = y, color_by = color_by, size_by = size_by, row_facet = row_facet, col_facet = col_facet
+      x = x,
+      y = y,
+      color_by = color_by,
+      size_by = size_by,
+      row_facet = row_facet,
+      col_facet = col_facet
     )
-    selector_list <- teal.transform::data_extract_multiple_srv(data_extract, data)
+
+    rule_diff <- function(other) {
+      function(value) {
+        if (!is.null(selector_list()[[other]]()$select)) {
+          othervalue <- selector_list()[[other]]()$select
+          if (identical(value, othervalue))
+            "Row and column facetting variables must be different."
+        }
+      }
+    }
+
+    selector_list <- teal.transform::data_extract_multiple_srv(
+      data_extract = data_extract,
+      datasets = data,
+      select_validation_rule = list(
+        x = ~ if (length(.) != 1) "Please select exactly one x var.",
+        y = ~ if (length(.) != 1) "Please select exactly one y var.",
+        color_by = ~ if (length(.) > 1) "There cannot be more than 1 color variable.",
+        size_by = ~ if (length(.) > 1) "There cannot be more than 1 size variable.",
+        row_facet = shinyvalidate::compose_rules(
+          shinyvalidate::sv_optional(),
+          ~ if (length(.) > 1) "There must be 1 or no row facetting variable.",
```

Comment:
See https://github.com/insightsengineering/teal.modules.general/pull/498/files#r1060582916 though in this case we don't validate others so maybe it makes sense to keep it like it is here

Reply:
done